cmd/entrypoint/main.go

```diff
@@ -134,9 +144,15 @@ func main() {
 			// same signature.
 			if status, ok := t.Sys().(syscall.WaitStatus); ok {
 				checkForBreakpointOnFailure(e, breakpointExitPostFile)
-				os.Exit(status.ExitStatus())
+				// ignore a step error i.e. do not exit if a container terminates with a non-zero exit code when onError is set to "continue"
+				if e.OnError != entrypoint.ContinueOnError {
+					os.Exit(status.ExitStatus())
+				}
+			}
+			// log and exit only if a step error must cause run failure
+			if e.OnError != entrypoint.ContinueOnError {
+				log.Fatalf("Error executing command (ExitError): %v", err)
```

Comment:
Checking if `onError` is set to `continue` to allow a step error. By default, the entrypoint considers a container exiting with non-zero as error and hence not included any special logic for `onError` set to `fail`. We can change this logic if needed.